src/admin/server.ts

```diff
@@ -1,29 +1,52 @@
 import { ApolloServer } from 'apollo-server-express';
 import { buildSubgraphSchema } from '@apollo/federation';
+import { Request } from 'express';
 import { typeDefsAdmin } from '../typeDefs';
 import { resolvers as adminResolvers } from './resolvers';
 import { sentryPlugin } from '@pocket-tools/apollo-utils';
 import {
   ApolloServerPluginLandingPageDisabled,
   ApolloServerPluginLandingPageGraphQLPlayground,
 } from 'apollo-server-core';
-import { client } from '../database/client';
-import s3 from '../aws/s3';
+import { ContextManager } from './context';
 
-export const server = new ApolloServer({
-  schema: buildSubgraphSchema([
-    { typeDefs: typeDefsAdmin, resolvers: adminResolvers },
-  ]),
-  plugins: [
-    sentryPlugin,
-    // Keep the settings we had when using v.2:
-    // no landing page on production + playground in other environments
-    process.env.NODE_ENV === 'production'
-      ? ApolloServerPluginLandingPageDisabled()
-      : ApolloServerPluginLandingPageGraphQLPlayground(),
-  ],
-  context: {
-    db: client(),
-    s3,
-  },
-});
+// Function signature for context creator; primarily for
+// injecting test contexts
+interface ContextFactory {
+  (req: Request): ContextManager;
+}
+
+/**
+ * Sets up and configures an ApolloServer for the application.
+ * @param contextFactory function for creating the context with
+ * every request
+ * @returns ApolloServer
+ */
+export function getServer(contextFactory: ContextFactory): ApolloServer {
```

Comment:
similar note here - would be nice to either import with an `as getAdminServer`, or rename this to `getAdminServer`.